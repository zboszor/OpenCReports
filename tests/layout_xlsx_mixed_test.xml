<?xml version="1.0"?>
<!DOCTYPE OpenCReport SYSTEM "opencreport.dtd">
<OpenCReport>
	<Datasources>
		<Datasource name="pandas" type="'pandas'" filename="test1.xlsx" />
	</Datasources>
	<Queries>
		<Query name="a" datasource="pandas" value="'Sheet1'" />
	</Queries>
	<Report>
		<ReportHeader>
			<Output>
				<Line>
					<literal>XLSX mixed values test</literal>
				</Line>
			</Output>
		</ReportHeader>
		<Detail>
			<FieldHeaders>
				<Output>
					<HorizontalLine size="1" bgcolor="'black'"/>
					<Line bgcolor="'0xe5e5e5'">
						<literal width="25">Value</literal>
						<literal width="1"/>
						<literal width="4">Text</literal>
						<literal width="1"/>
						<literal width="7">Numeric</literal>
						<literal width="1"/>
						<literal width="8">Datetime</literal>
						<literal width="1"/>
						<literal width="4">Null</literal>
					</Line>
					<HorizontalLine size="1" bgcolor="'black'"/>
					<HorizontalLine size="4" bgcolor="'white'"/>
				</Output>
			</FieldHeaders>
			<FieldDetails>
				<Output>
					<Line bgcolor="iif(r.detailcnt%2,'0xe5e5e5','white')">
						<field value="value" width="25" format="isnumeric(r.value) ? '%.4d' : nulls()" align="isnumeric(r.value) ? 'right' : 'left'"/>
						<literal width="1"/>
						<field value="isstring(value) ? '*' :  ' '" width="4" align="center"/>
						<literal width="1"/>
						<field value="isnumeric(value) ? '*' :  ' '" width="7" align="center"/>
						<literal width="1"/>
						<field value="isdatetime(value) ? '*' :  ' '" width="8" align="center"/>
						<literal width="1"/>
						<field value="isnull(value) ? '*' :  ' '" width="4" align="center"/>
					</Line>
				</Output>
			</FieldDetails>
		</Detail>
	</Report>
</OpenCReport>
