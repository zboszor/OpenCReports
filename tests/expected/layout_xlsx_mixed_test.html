<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8" >
<link rel="preconnect" href="//fonts.gstatic.com/" >
<link rel="preconnect" href="//fonts.gstatic.com/" crossorigin >
<link rel="stylesheet" type="text/css" href="//fonts.googleapis.com/css?family=Courier" >
<style>
@import url('//fonts.googleapis.com/css?family=Courier'); 
body { background-color: #ffffff; }
table { border: 0; border-spacing: 0; padding: 0; width:100%; }
p { margin: 0; padding: 0; }
span { display: inline-flex; flex-wrap: nowrap; white-space: nowrap; text-overflow: clip; overflow: hidden; vertical-align: middle; margin: 0; padding: 0; }
</style>
<title>OpenCReports report</title></head>
<body>
<!--part start-->
<table>
<!--part row start-->
<tr style="display: flex; align-items: flex-start; ">
<!--part column start-->
<td style="width: 566.48pt; text-overflow: clip; ">
<!--part column end-->
</td>
<!--part row end-->
</tr>
<!--part row start-->
<tr style="display: flex; align-items: flex-start; ">
<!--part column start-->
<td style="width: 100%; ">
<!--part column end-->
</td>
<!--part row end-->
</tr>
<!--part row start-->
<tr style="display: flex; align-items: flex-start; ">
<!--part column start-->
<td style="width: 566.48pt; text-overflow: clip; ">
<!--output start--><section style="clear: both; width: 566.48pt; ">
<p><span style="line-height: 23.00pt; width: 566.48pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">XLSX&nbsp;mixed&nbsp;values&nbsp;test</span></p>
<!--output end--></section>
<!--output start--><section style="clear: both; width: 566.48pt; ">
<div style="display: block; margin-left: 0.00pt; height: 1.00pt; width: 566.48pt; background-color: #000000; ">&nbsp;</div>
<p><span style="line-height: 23.00pt; width: 180.00pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">Value</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 28.80pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">Text</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 50.40pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">Numeric</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 57.60pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">Datetime</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 28.80pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">Null</span></p>
<div style="display: block; margin-left: 0.00pt; height: 1.00pt; width: 566.48pt; background-color: #000000; ">&nbsp;</div>
<div style="display: block; margin-left: 0.00pt; height: 4.00pt; width: 566.48pt; background-color: #ffffff; ">&nbsp;</div>
<!--output end--></section>
<!--output start--><section style="clear: both; width: 566.48pt; ">
<p><span style="line-height: 23.00pt; width: 180.00pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">Some&nbsp;text</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 28.80pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">*</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 50.40pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 57.60pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 28.80pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span></p>
<!--output end--></section>
<!--output start--><section style="clear: both; width: 566.48pt; ">
<p><span style="line-height: 23.00pt; width: 180.00pt; justify-content: end; text-align: end; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">65.3000</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 28.80pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 50.40pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">*</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 57.60pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 28.80pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">&nbsp;</span></p>
<!--output end--></section>
<!--output start--><section style="clear: both; width: 566.48pt; ">
<p><span style="line-height: 23.00pt; width: 180.00pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">10/19/25&nbsp;10:50:37</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 28.80pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 50.40pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 57.60pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">*</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 28.80pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span></p>
<!--output end--></section>
<!--output start--><section style="clear: both; width: 566.48pt; ">
<p><span style="line-height: 23.00pt; width: 180.00pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 28.80pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">*</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 50.40pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 57.60pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 28.80pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #ffffff; color: #000000; ">*</span></p>
<!--output end--></section>
<!--output start--><section style="clear: both; width: 566.48pt; ">
<p><span style="line-height: 23.00pt; width: 180.00pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">Some&nbsp;text&nbsp;2</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 28.80pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">*</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 50.40pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 57.60pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 7.20pt; justify-content: start; text-align: start; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span><span style="line-height: 23.00pt; width: 28.80pt; justify-content: center; text-align: center; font-family: 'Courier'; font-size: 12.00pt; background-color: #e5e5e5; color: #000000; ">&nbsp;</span></p>
<!--output end--></section>
<!--part column end-->
</td>
<!--part row end-->
</tr>
<!--part row start-->
<tr style="display: flex; align-items: flex-start; ">
<!--part column start-->
<td style="width: 100%; ">
<!--part column end-->
</td>
<!--part row end-->
</tr>
<!--part end-->
</table>
</body></html>
